### PR TITLE
W-12008687 : update isValid with query-config api

### PR DESCRIPTION
**Pull request by rajeshkrs**

### CODE REVIEW

File: src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceConnection.java [hunk 1]
```diff
@@ -457,4 +463,20 @@ public String getTenantUrl() {
...
         this.tenantUrl = tenantUrl;
     }
+
+    private QueryExecutor createQueryExecutor() {
+        return new QueryExecutor(this);
+    }
+
+    QueryConfigResponse getQueryConfigResponse() throws SQLException {
+        try {
+            QueryExecutor executor = createQueryExecutor();
```
Comment (soaggarwal) [inline src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceConnection.java:473]: We should not make call to hyper if enableStreamFlow is false in connectionProperties

File: src/test/java/com/salesforce/cdp/queryservice/core/QueryServiceConnectionTest.java [hunk 2]
```diff
@@ -252,25 +244,17 @@ public void testGrpcIsValid() throws SQLException {
         assertThat(connection.isValid(10)).isFalse();
     }
 
-    @Test
```
Comment (soaggarwal) [inline src/test/java/com/salesforce/cdp/queryservice/core/QueryServiceConnectionTest.java:255]: Please add the testcase

File: src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceConnection.java [hunk 3]
```diff
@@ -351,17 +362,12 @@ public boolean isValid(int timeout) throws SQLException {
         }
 
         try {
-            PreparedStatement statement = this.prepareStatement(TEST_CONNECT_QUERY);
-            return statement.execute();
+            QueryConfigResponse configResponse = getQueryConfigResponse();
```
Comment (soaggarwal) [inline src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceConnection.java:365]: Please test retry on failure

File: src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceConnection.java (hunk 1 above)
Comment (rajeshkrs) [reply src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceConnection.java:473]: I am passing `enable-stream-flow` in header and will make changes in QS for this.
https://github.com/rajeshkrs/Salesforce-CDP-jdbc/pull/4/files#diff-df23f4d023487b3e58cd1dce72efdd280ba081882391219cedcea789a512ed69R113

File: src/test/java/com/salesforce/cdp/queryservice/core/QueryServiceConnectionTest.java (hunk 2 above)
Comment (rajeshkrs) [reply src/test/java/com/salesforce/cdp/queryservice/core/QueryServiceConnectionTest.java:255]: this test case was for exception case when hyper threw error and then we called trino endpoint in catch block.
that is not valid now, so I removed the test case.

**Comment by rajeshkrs** [comment]
merged it in main repo

**Pull request closed without merging**